### PR TITLE
[Automation] Quality of Life improvements

### PR DESCRIPTION
**Pull request by sophiaaar**
**This PR is a collection of automation "quality of life" work that I worked on during graphics hackweek**

### Purpose of this PR
* GitHub Actions bot will comment on a PR (if it is the first PR that user has made to this repo) and welcome them to the repository
* GitHub Actions will add labels to the PR based on which files changed
* Since labelling now happens automatically for SRP packages, that section of the PR template was removed
* SRP ABV on Yamato will be started automatically when a PR is created (note: this currently also happens when a draft PR is created. A feature request has been made to only start the jobs if the PR is marked as "ready for review")
* Tests will be automatically started based on the files changes when pushing to a branch (eg if you make a change to the Shader Graph package, the 'All ShaderGraph CI' job will start)
* Where appropriate, jobs with no commands were switched to "virtual jobs" (ie the VM was removed to not waste resources)
* Rider package version was updated to stop the warning message in the logs
* Linux builders updated to stable
* `## [Unreleased]` added to changelogs so we can keep them up-to-date with changes without failing the package validation suite

---
### Testing status

**Yamato**: 
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline/tree/automation%252Fgithub-actions

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  - TestProjects/HDRP*/**/*
+  
+Universal:
+  - com.unity.render-pipelines.universal/**/*
+  - com.unity.render-pipelines.lightweight/**/*
+  - TestProjects/Universal*/**/*
+  
+Shader Graph:
+  - com.unity.shadergraph/**/*
+  - TestProjects/ShaderGraph*/**/*
```
Comment (phi-lira) [inline .github/labeler.yml]: We should add to ShaderGraph the intergration ShaderGraph files inside URP and HDRP, f.ex, in URP this sits in `com.unity.render-pipelines.universal\Editor\ShaderGraph`

File: .yamato/upm-ci-abv.yml [hunk 2]
```diff
@@ -75,11 +65,7 @@ all_project_ci_nightly:
   triggers:
     recurring:
       - branch: master
-        frequency: daily
-  artifacts:
```
Comment (phi-lira) [inline .yamato/upm-ci-abv.yml]: Why is this removed?

File: .yamato/upm-ci-shadergraph.yml [hunk 3]
```diff
@@ -261,6 +255,15 @@ All_{{ project.name }}_{{ editor.version }}:
     
   {% endfor %}
   {% endfor %}
+  {% if editor.version == "fast-trunk" %}
+  triggers:
+    changes:
+      only:
+        - "com.unity.shadergraph/**/*"
+        - "TestProjects/ShaderGraph*/**/*"
```
Comment (phi-lira) [inline .yamato/upm-ci-shadergraph.yml]: Same as above here

File: .yamato/upm-ci-shadergraph_stereo.yml [hunk 4]
```diff
@@ -66,18 +66,21 @@ projects:
...
   {% for testplatform in testplatforms %}
     - .yamato/upm-ci-{{ project.name | downcase }}.yml#{{ project.name }}_{{ platform.name }}_{{ testplatform.name }}_{{ editor.version }}
   {% endfor %}
   {% endfor %}
+  {% if editor.version == "fast-trunk" %}
+  triggers:
+    changes:
+      only:
+        - "com.unity.shadergraph/**/*"
+        - "TestProjects/ShaderGraph*/**/*"
```
Comment (phi-lira) [inline .yamato/upm-ci-shadergraph_stereo.yml]: Same as above. Need to add universal shader graph editor.

File: .github/labeler.yml (hunk 1 above)
Comment (sophiaaar) [reply .github/labeler.yml]: good idea, thank you

File: .yamato/upm-ci-abv.yml (hunk 2 above)
Comment (sophiaaar) [reply .yamato/upm-ci-abv.yml]: ah that was a mistake - the artifacts were intentional but not the frequency

Comment (sophiaaar) [reply .yamato/upm-ci-abv.yml]: oh sorry no, it wasnt removed - see line 68

**Comment by sebastienlagarde** [comment]
Love the Auto Label.

>SRP ABV on Yamato will be started automatically when a PR is created 
Guess in current context it is dangerous? Or maybe I don't understand it correctly.
If I create a PR for a comment fix , then I will trigger ABV automatically?

**Comment by sophiaaar** [comment]
> > SRP ABV on Yamato will be started automatically when a PR is created
> Guess in current context it is dangerous? Or maybe I don't understand it correctly.
> If I create a PR for a comment fix , then I will trigger ABV automatically?

Yeah I know that this could potentially be overkill, but we can use our best judgement to cancel jobs if we think running a full ABV is unnecessary. My thinking was that those cases would be quite rare :)

**Comment by sebastienlagarde** [comment]
> > > SRP ABV on Yamato will be started automatically when a PR is created
> > > Guess in current context it is dangerous? Or maybe I don't understand it correctly.
> > > If I create a PR for a comment fix , then I will trigger ABV automatically?
> 
> Yeah I know that this could potentially be overkill, but we can use our best judgement to cancel jobs if we think running a full ABV is unnecessary. My thinking was that those cases would be quite rare :)

They are frequent :). I can point you on 3 to 4 PR the last week. Indeed not a lot but if we want people to cancel the build, they need to know there is a build. Is there a way to provide link to the yamato for this branch in the PR automaticalyl, so at least we can click on it when creating the PR to cancel the build?

**Comment by sophiaaar** [comment]
> > > > SRP ABV on Yamato will be started automatically when a PR is created
> > > > Guess in current context it is dangerous? Or maybe I don't understand it correctly.
> > > > If I create a PR for a comment fix , then I will trigger ABV automatically?
> > 
> > 
> > Yeah I know that this could potentially be overkill, but we can use our best judgement to cancel jobs if we think running a full ABV is unnecessary. My thinking was that those cases would be quite rare :)
> 
> They are frequent :). I can point you on 3 to 4 PR the last week. Indeed not a lot but if we want people to cancel the build, they need to know there is a build. Is there a way to provide link to the yamato for this branch in the PR automaticalyl, so at least we can click on it when creating the PR to cancel the build?

Yep, they get added to the Status part (above the comment box) :)

**Comment by sebastienlagarde** [comment]
> > > > SRP ABV on Yamato will be started automatically when a PR is created
> > > > Guess in current context it is dangerous? Or maybe I don't understand it correctly.
> > > > If I create a PR for a comment fix , then I will trigger ABV automatically?
> > 
> > 
> > Yeah I know that this could potentially be overkill, but we can use our best judgement to cancel jobs if we think running a full ABV is unnecessary. My thinking was that those cases would be quite rare :)
> 
> They are frequent :). I can point you on 3 to 4 PR the last week. In particular, doc only PR. Indeed not a lot but if we want people to cancel the build, they need to know there is a build. Is there a way to provide link to the yamato for this branch in the PR automaticalyl, so at least we can click on it when creating the PR to cancel the build?
> 
> Maybe we should do a case for documentation btw? i.e identify if change are only in the documentation~ folder, changelog or readme.

**A feature request has been made to only start the jobs if the PR is marked as "ready for review"**
This currently will be the best solution, only trigger ABV if this is marked as it

**Comment by sebastienlagarde** [comment]
Nice works btw, thanks for this PR!

**Comment by sophiaaar** [comment]
> Maybe we should do a case for documentation btw? i.e identify if change are only in the documentation~ folder, changelog or readme.

I think this is possible actually, yeah! Let me see what i can do
